application/comit_node/src/ledger_query_service/timer_poll_future.rs

```diff
@@ -0,0 +1,217 @@
+use futures::{
+    future::Future,
+    stream::{iter_ok, Stream},
+};
+use std::time::{Duration, Instant};
+use tokio::timer::Interval;
+
+#[allow(dead_code)]
+fn poll_dedup<
+    I: PartialEq + Clone + 'static + Send,
+    E: Send + 'static,
+    Fut: Future<Item = Vec<I>, Error = E> + Send + 'static,
+    F: 'static + Send + FnMut() -> Fut,
+    Ticker: Stream<Item = (), Error = ()> + Send + 'static,
+>(
+    ticker: Ticker,
+    mut f: F,
+) -> Box<dyn Stream<Item = I, Error = E> + Send + 'static> {
+    let mut seen = vec![];
+
+    Box::new(
+        ticker
+            .map_err(|_| unreachable!("Ticker cannot error"))
+            .and_then(move |_| f())
+            .map(iter_ok)
+            .flatten()
+            .filter(move |item| {
+                let is_new = !seen.contains(item);
+
+                if is_new {
+                    seen.push(item.clone());
+                }
+                is_new
+            }),
+    )
+}
+
+/// Keep polling a future that returns a vec and puts the results
+/// deduplicated results into a stream.
+// Here in case we want to use it later
+#[allow(dead_code)]
+pub fn poll_future_into_stream<
```

Comment:
Note: this is here in case we want to be able to return query results as a stream (before implementing websockets).